### ISSUE TITLE
VVVVVV stops drawing for me when switching workspaces (Xubuntu 18.04, xfwm4)

### ISSUE THREAD

**Issue by Dav999-v** [TerryCavanagh/VVVVVV#283]
@InfoTeddy asked me to open an issue about this, so I'll try to describe everything I know. Of everyone on the VVVVVV Discord it only seems to affect me, while there are enough people there running VVVVVV on Linux, so it probably has something to do with my window manager, xfwm4. I'm on Xubuntu 18.04. It also only happens to me in VVVVVV and VVVVVV-CE, and, as `git bisect` revealed, only after commit c322ae131e599562aee019ebec0fbc384fed0095.

When VVVVVV is in windowed mode, and I either switch to a different workspace, or collapse and re-expand the window (collapsing only leaves the top bar), VVVVVV most of the time completely stops drawing and just leaves whatever was last on screen in its location:

![](https://cdn.discordapp.com/attachments/665284975905603584/718124431364325426/unknown.png)

The game clearly hasn't frozen, since sounds still play. Another issue is that VVVVVV pulls me back to its workspace every time I click on a different workspace in my panel while VVVVVV is in focus on the current workspace. After I click on VVVVVV's workspace, it also does the same thing when I use keyboard shortcuts to move to a different workspace (ctrl+F1-F5, or ctrl+alt+left/right), but only once.

I can randomly cause the game to completely restore drawing by repeatedly switching workspaces back and forth or by repeatedly collapsing and expanding the window.

Simply unfocusing VVVVVV (getting a pause screen) and working with other windows on the same workspace causes no problems. In fact, when VVVVVV is not in focus and I switch workspaces or collapse/expand, the problem seems rare to non-existent.

While the window contents are frozen, even alt+enter to go to fullscreen, or the "toggle fullscreen" button in the graphic options, do not work (the menu toggle does still make a squeak sound and can still be repeatedly toggled, but the game stays windowed). After trying to switch to fullscreen, it does suddenly go into fullscreen when I switch workspaces back and forth once, or collapse and re-expand the window once.

Simply resizing the window does not fix the problem, but maximizing it does. In fact, I just found out the problem doesn't occur while the window is maximized, but it does still pull me back into its workspace when I try to switch away from it like I described above.

**Comment by flibitijibibo** [comment]
When @InfoTeddy and I were originally talking about this the consensus was to replicate FNA's window behavior, but upon revisiting the event loop it turns out I already did that... like, 6 years ago. So at first I just reverted the above commit since it was exactly FNA's behavior and that's what I know works, but then I looked at ResizeScreen just to be sure and now I know why it _wasn't_ working: we would get a focus event and call SetWindowFullscreen, at which point a resize event would come in and cause a fullscreen toggle that would often undo what the event loop did.

Also there was a minor issue in that I didn't even copy my own work correctly and the toggles were swapped

_Anyway_

I've reverted the commit from the OP and fixed it for real this time, now it _actually_ behaves more like an FNA game with AllowUserResizing enabled (mostly):

https://github.com/TerryCavanagh/VVVVVV/commit/a15d01ad80066428f8dff846a6e3188c45f78ac0

**Comment by Dav999-v** [comment]
Yep, looks like it's now fixed. Thanks!
